railgun/internal/cmd/rootcmd/servers.go

```diff
@@ -45,23 +46,33 @@ func StartAdmissionServer(ctx context.Context, c *manager.Config) error {
 	return nil
 }
 
-func StartProfilingServer(ctx context.Context, c *manager.Config) error {
+func StartDiagnosticsServer(ctx context.Context, port int, c *manager.Config) (diagnostics.Server, error) {
 	deprecatedLogger, err := util.MakeLogger(c.LogLevel, c.LogFormat)
 	if err != nil {
-		return err
+		return diagnostics.Server{}, err
 	}
 	logger := logrusr.NewLogger(deprecatedLogger)
 
-	if !c.EnableProfiling {
-		logger.Info("profiling server disabled")
-		return nil
+	if !c.EnableProfiling && !c.EnableConfigDumps {
+		logger.Info("diagnostics server disabled")
+		return diagnostics.Server{}, nil
 	}
 
-	s := diagnostics.Server{Logger: logger}
+	s := diagnostics.Server{
+		Logger:           logger,
+		ProfilingEnabled: c.EnableProfiling,
+	}
+	if c.EnableConfigDumps {
+		s.ConfigDumps = util.ConfigDumpDiagnostic{
+			DumpsIncludeSensitive: c.DumpSensitiveConfig,
+			SuccessfulConfigs:     make(chan file.Content),
```

Comment:
Do you really needs two channel for the same thing ? file.Content struct has nested objects within it.
concise struct as following might also resolve your problem.

type struct{
  ok: bool
  config: chan file.Content.
}

Reply:
Not really, choice was more or less arbitrary. Not sure how much resources they consume, but the struct feels more lightweight, so d669d6b8